addons/account/models/account_move.py

```diff
@@ -1914,8 +1935,10 @@ def post(self):
             if not move.invoice_date and move.is_invoice(include_receipts=True):
                 move.invoice_date = fields.Date.context_today(self)
                 move.with_context(check_move_validity=False)._onchange_invoice_date()
+            if move.company_id.tax_lock_date and move.date <= move.company_id.tax_lock_date:
+                move.date = move.company_id.tax_lock_date + timedelta(days=1)
+                move.with_context(check_move_validity=False)._onchange_currency()
```

Comment:
check_move_validity=False?

Reply:
Auto-balance of lines will be triggered performing some writes. However the move will be unbalanced until the end of the method (we are not in onchange mode here). Don't worry, the check_balanced is done at the end.

Reply:
add a comment to explain that